File: IOS/LuaViewSDK/Lua/Net/LuaRequest.m
//
//  LuaRequest.m
//  LuaViewSDK
//
//  Created by 王金东 on 17/2/6.
//  Copyright © 2017年 王金东. All rights reserved.
//

#import "LuaRequest.h"
#import "NetworkManager.h"
#import "LSCValue.h"

@implementation LuaRequest

+ (void)post:(NSString *)url params:(NSDictionary *)params success:(LSCFunction *)success fail:(LSCFunction *)fail {
    [NetworkManager post:url params:params callBack:^(BOOL s, NSDictionary *result) {
        NSMutableArray *luaArray = [NSMutableArray array];
        [luaArray addObject:[LSCValue objectValue:result]];
        if (s) {
            [success invokeWithArguments:luaArray];
        }else{
            [fail invokeWithArguments:luaArray];
        }
    }];
}
@end

File: Android/LuaViewSDK/app/src/main/java/com/core/lua/net/LuaRequest.java
package com.core.lua.net;

import com.core.lua.local.net.NetworkInterface;
import com.core.lua.local.net.NetworkManager;

import java.util.Map;

import cn.vimfung.luascriptcore.LuaFunction;
import cn.vimfung.luascriptcore.LuaValue;
import cn.vimfung.luascriptcore.modules.oo.LuaObjectClass;

/**
 * Created by JD on 2017/2/6.
 */
public class LuaRequest extends LuaObjectClass {

    public static void post(String url,Map params,final LuaFunction success,final LuaFunction fail){
        System.out.println("post");
        NetworkManager.post(url, params, new NetworkInterface.NetworkCallBack() {
            @Override
            public void callBack(boolean s, Map result) {
                LuaValue[] resultValue = new LuaValue[1];
                resultValue[0] = new LuaValue(result);
                if (s){
                    if (success != null) {
                        success.invoke(resultValue);
                    }
                }else{
                    if (fail != null) {
                        fail.invoke(resultValue);
                    }
                }
            }
        });
    }
}
